generators/client/templates/angular/src/main/webapp/app/admin/audits/audits.component.ts.ejs

```diff
@@ -30,113 +29,110 @@ import { AuditsService } from './audits.service';
   selector: '<%= jhiPrefixDashed %>-audit',
   templateUrl: './audits.component.html'
 })
-export class AuditsComponent implements OnInit, OnDestroy {
-    audits: Audit[];
-    fromDate: string;
-    itemsPerPage: any;
-    links: any;
-    page: number;
-    routeData: any;
-    predicate: any;
-    previousPage: any;
-    reverse: boolean;
-    toDate: string;
-    totalItems: number;
+export class AuditsComponent implements OnInit {
+    audits: Audit[] = [];
+    fromDate = '';
+    itemsPerPage = ITEMS_PER_PAGE;
+    page!: number;
+    predicate!: string;
+    previousPage!: number;
+    ascending!: boolean;
+    toDate = '';
+    totalItems = 0;
+    canLoad = false;
+    initialized = false;
+    dataHaveBeenLoaded = false;
```

Comment:
I'm not a big fan of those 3 booleans... Can we instead use a `state` variable with 3 values `'initialized' | 'loading' | 'loaded'` or something like this ? WDYT ?

Reply:
Thanks for pointing this out! I removed all 3 variables:
* canLoad --> now it's function, value is calculated
* initialized --> just removed, filter section survives if this is shown before `ngOnInit`
* dataHaveBeenLoaded --> `audits` array variable can now be also `null`, null means that data not loaded yet

I also removed `eslint disable` lines and set `@typescript-eslint/no-non-null-assertion` off as decided in #10639 - hopefully Git conflicts will not arise if merging different PR-s (exactly the same change in `.eslintrc`).

I amended previous review commit and force pushed.